### DISCUSSION TITLE
Effects of running application with and without the Java Agent

### DISCUSSION THREAD

**Discussion by mossinipaolo** [open-telemetry/opentelemetry-java-instrumentation#10890] [Q&A] [answered]
Hi all!
I'm quite new to OpenTelemetry, and I'm trying to attach the Java agent to my application, and then to add more detail to my traces by adding instrumentation to my application's business logic using the OpenTelemetry API.

I would like to run my application with or without the Java agent depending on the deployment.
And I have a doubt about this scenario: what happens to my manual instrumentation code that I added in my business logic when I run the application without the Java agent? Will I have NullPointerException all over my code?

Is there a way to disable tracing without adding logic to check if the agent is running?

Thanks in advance

**Comment by laurit** [comment, marked as answer]
When using agent you get access to `OpenTelemetry` instance by calling `GlobalOpenTelemetry.get()`. If the agent is not running and you have not configured a global `OpenTelemetry` instance yourself it will return a no-op instance.
